en/docs/develop/adding-scim2-custom-user-schema-support.md

```diff
@@ -0,0 +1,528 @@
+# Adding SCIM2 Custom User Schema Support
+
+WSO2 Identity Server allows adding custom attributes into user objects through [Enterprise User Extension](../../develop/extending-scim2-user-schemas). From 5.12.0 onwards, 
+you can use this custom schema to add custom attributes of the user.
+
+**Reasons why we introduced custom schema to add custom attributes:**
+1. According to the current model,  Enterprise User Extension attributes are in a file and those configurations are applicable in the server level.
+2. Enterprise User Extension is a defined schema and it should not be allowed to modify it.
+
+!!! Note 
+    SCIM2 Custom User Schema Support is added by default in WSO2 Identity Server
+    version 5.12.0 onwards.
+    
+### Add custom schema dialect
+
+1. Login to Management console as admin/admin credentials
+2. Add the custom schema dialect if it is not found
+    - Click **Add** under **Claims**
+    - Click on **Add Claim Dialect** .
+    - Add **urn:scim:custom:schema** claim dialect
+    
+    ![add-scim-custom-schema-dialect](../assets/img/develop/add-custom-schema-dialect.png)
+    
+    - Click **Add**
+    
+
+### Add custom local claim
+
+1. Click **Add** under **Claims**.
+2. Click **Add Local Claim** and enter the following values. 
+    - **Claim URI**: http://wso2.org/claims/customclaim
+    - **Display Name**: Custom Claim
+    - **Description**: Custom Claim
+    - **Mapped Attribute(s**):
+        - **User Store Domain Name**: PRIMARY
+        - **Mapped Attribute**: customClaim
+    - **Supported By Default**: True
+
+    ![add-scim-local-custom-claim](../assets/img/develop/add-custom-local-claim.png)
+    
+    If you want add any additional properties for the scim attribute, you can add it using **Additional Properties** 
+    in the claim configuration.
+    
+    Following additional properties can be added to the custom schema via **Additional Properties** element.
+       
+        - dataType
+        - multiValued
+        - required
+        - caseExact
+        - mutability
+        - returned
+        - uniqueness
+        - subAttributes
+        - referenceTypes
+        
+    Eg: dataType: boolean
+    
+ 
+!!! note
+    Note the following:
+
+    - You can use the word "customClaim" (or any other preferred word) as the **Mapped Attribute** only when using a JDBC user store because JDBC user stores will automatically create a new attribute if it does not already exist in the user store. However, If you are using LDAP or Active Directory, you will have to use an attribute that exists in the user store already.
```

Comment:
```suggestion
    - You can use the word `customClaim` or any other preferred word as the **Mapped Attribute** when using a JDBC userstore because JDBC userstores will automatically create a new attribute if it does not already exist in the userstore. However, if you are using LDAP or Active Directory, you will have to use an attribute that exists in the userstore already.
```